cli_tools/common/utils/logging/service/log_entry.go

```diff
@@ -171,27 +190,22 @@ type OutputInfo struct {
 	FailureMessageWithoutPrivacyInfo string `json:"failure_message_without_privacy_info,omitempty"`
 }
 
-func (l *Logger) updateParams(projectPointer *string) {
+func (l *Logger) updateParams() {
 	l.mutex.Lock()
 	defer l.mutex.Unlock()
 
-	if projectPointer == nil {
-		return
-	}
+	l.updateProject()
+}
 
-	project := *projectPointer
-	obfuscatedProject := Hash(project)
+func (l *Logger) updateProject() {
+	if commonParams := l.Params.commonParams(); commonParams != nil {
+		if commonParams.UpdatableProject == nil {
+			return
+		}
+		project := commonParams.UpdatableProject.StringValue()
+		obfuscatedProject := Hash(project)
 
-	if l.Params.ImageImportParams != nil {
-		l.Params.ImageImportParams.CommonParams.Project = project
-		l.Params.ImageImportParams.CommonParams.ObfuscatedProject = obfuscatedProject
-	}
-	if l.Params.ImageExportParams != nil {
-		l.Params.ImageExportParams.CommonParams.Project = project
-		l.Params.ImageExportParams.CommonParams.ObfuscatedProject = obfuscatedProject
-	}
-	if l.Params.InstanceImportParams != nil {
-		l.Params.InstanceImportParams.CommonParams.Project = project
-		l.Params.InstanceImportParams.CommonParams.ObfuscatedProject = obfuscatedProject
+		commonParams.Project = project
+		commonParams.ObfuscatedProject = obfuscatedProject
```

Comment:
`UpdatableProject` has a bad code smell to me.
1. From its name, it sounds mutable. Is that what you want?
2. Who is responsible for initializing it? Is it really okay that the value is ever nil?

The approach that I would have expected is to have the 'project' value created and validated once, and then made immutable. Are there limitations that make that impossible?